tests/unit/test_global_correction.py

```diff
@@ -106,3 +109,84 @@ def test_lhc_global_correct_methods(tmp_path, model_inj_beams, method):
+
+@pytest.mark.basic
+def test_update_response(tmp_path, model_inj_beams):
+    """ Tests if the response is updated. """
+    # create the accelerator instance
+    beam = model_inj_beams.beam
+    knob = "kqd.a78"#f'kq10.l1b{beam}'
+    accel_inst = manager.get_accelerator(model_inj_beams)
+    delta = tfs.TfsDataFrame(
+        3.34e-05,
+        index=[knob], 
+        columns=[DELTA]
+        )
+    optics_params = ['PHASEX', 'PHASEY']
+
+    # Get a response dict from the model
+    ref_resp_dict = response_madx.create_fullresponse(accel_inst, [knob])
+
+    corr_model_path = tmp_path / f"twiss_cor{EXT}"
+    same_model = _create_corrected_model(corr_model_path, [], accel_inst, False)
+    
+    # As orbit_dpp is not in the response, it should not be updated. First for response_madx
+    new_resp_dict = _update_response(
+        accel_inst=accel_inst,
+        corrected_elements=same_model,
+        optics_params=optics_params,
+        corr_files=[],
+        variable_categories=[knob],
+        update_dpp=False,
+        update_response="madx"
+)
+
+    # Check everything is the same, as model has not changed.
+    for key in ref_resp_dict.keys():
+        assert_frame_equal(ref_resp_dict[key], new_resp_dict[key])
+
+    corr_file = tmp_path / "corr.madx"
+    with open(corr_file, "w") as f:
+        f.write(f"{knob} = {knob}{delta.loc[knob, DELTA]:+.16e};")
+
+    corr_model = _create_corrected_model(corr_model_path, [corr_file], accel_inst, False)
+    assert not same_model.equals(corr_model)
+
+    # Now for response_madx with the model changed
+    new_resp_dict = _update_response(
+        accel_inst=accel_inst,
+        corrected_elements=corr_model,
+        optics_params=optics_params,
+        corr_files=[corr_file],
+        variable_categories=[knob],
+        update_dpp=False,
+        update_response="madx"
+    )
+    for key in new_resp_dict.keys():
+        # If the original response is 0, there is no reason why the new one should not be.
+        if not ref_resp_dict[key][knob].sum() == 0:
+            # The values are not the same, as the model has changed.
+            assert not ref_resp_dict[key].equals(new_resp_dict[key])
+
+            # Check the general structure of the response
+            assert new_resp_dict[key].columns.equals(ref_resp_dict[key].columns)
+            assert new_resp_dict[key].index.equals(ref_resp_dict[key].index)
+
+    # Now for response_twiss
+    new_resp_dict = _update_response(
+        accel_inst=accel_inst,
+        corrected_elements=corr_model,
+        optics_params=optics_params,
+        corr_files=[],
+        variable_categories=[knob],
+        update_dpp=False,
+        update_response=True
+    )
+    for key in new_resp_dict.keys(): # The new dict will only have the keys for the selected optics_params
+        # The values are not the same, as the model has changed.
+        assert not ref_resp_dict[key].equals(new_resp_dict[key])
```

Comment:
I think this is not a good test, as ref_resp was created with madx. You should create a ref also with the twiss repsonse and original model first.